File: 05. Average/Average.m
img = imread("Galaxy.tif");
img = im2double(img);
sum = zeros(size(img));
mean = 0;
sigma = 0.1;
for i = 1:100
	noise = normrnd(mean, sigma, size(img));
	noise = img + noise;
	sum = sum + noise;
end
avg = sum/i;
subplot(131), Imshow(img), subplot(132), imshow(noise), subplot(133), imshow(avg);
	

File: 05. Average/Average.py
import cv2
import numpy as np
from matplotlib import pyplot as plt

#IMAGE AVERAGES

img=cv2.imread("Galaxy.tif",0)
img=np.float64(img)
sum_img=np.zeros((img.shape),np.float64)
mean=0
sigma=20
for i in range(100):
	noise = np.random.normal(mean,sigma,(img.shape))
	noise = img+noise
	sum_img = sum_img + noise

avg = sum_img/i

plt.subplot(131),plt.imshow(img,'gray')
plt.subplot(132),plt.imshow(noise,'gray')
plt.subplot(133),plt.imshow(avg,'gray')
plt.show()
